driver-core/src/test/java/com/datastax/driver/core/DirectCompressionTest.java

```diff
@@ -29,6 +29,7 @@ public class DirectCompressionTest extends CompressionTest {
    */
   @Test(groups = "short")
   public void should_function_with_snappy_compression() throws Exception {
+    skipTestWithCassandraVersionOrHigher("4.0.0", "snappy");
```

Comment:
I'm not sure I understand why this test isn't applicable to Cassandra 4.x.  That version still supports Snappy compression doesn't it?

Reply:
It's strange because 4.x will definitely [return an error message saying it's not supported in v5](https://github.com/apache/cassandra/blob/2df88755022f09ecef86b2173a0282e1586cd7b2/src/java/org/apache/cassandra/transport/messages/StartupMessage.java#L90-L99)... but it looks like it's [still an option](https://github.com/datastax/native-protocol/blob/1.x/src/main/resources/native_protocol_v5.spec#L863-L869).

Huh... actually, scratch that.  It looks like the v5 protocol in native-protocol is out of date.  Based on the version of the spec in-tree [Snappy isn't supported anymore](https://github.com/apache/cassandra/blob/trunk/doc/native_protocol_v5.spec#L201-L213).

Huh... TIL.

Looks like our v5 spec is out-of-date.

Reply:
Filed an [issue in native-protocol](https://github.com/datastax/native-protocol/issues/46) to fix that